### ISSUE TITLE
CUPS displays PPD options without "*OpenUI" ... "*CloseUI"

### ISSUE THREAD

**Issue by michaelrsweet** [michaelrsweet/testcups#7]
Version: 1.1-current
Originally Reported By: till.kamppeter
Originally Reported On: 2003-04-04 @ 11:25:00

**Comment by michaelrsweet** [comment]
mike on 2003-04-09 @ 11:55:14

Please read the Adobe PPD specification.

Certain options defined in the PPD spec are explicitly allowed to omit the OpenUI/CloseUI stuff.  The Resolution option is among those defined in the spec.  Thus, the previous behavior of omitting option keywords that were not inside OpenUI/CloseUI was in error, and the current CVS is correct.

**Comment by michaelrsweet** [comment]
till.kamppeter on 2003-04-14 @ 14:57:28

I have checked the PPD specs and found out that the &quot;*Resolution ...&quot; lines are superflous, the &quot;*DefaultResolution: ...&quot; is enough to tell to a client application what the printer&#039;s resolution is. So I removed the &quot;*Resolution ...&quot; lines from the &lt;ppdentry&gt; sections of the Foomatic XML files so that they do not get into the PPDs any more. This lets the 1-choice &quot;Resolution&quot; widgets disappear.

See also Adobe PPD spec 4.3, section 5.9, p. 84, Note and section 4.5, p. 40.
